### PR TITLE
Encapsulate compute_ssst_frf() in a separate function

### PR DESCRIPTION
**Pull request by ppoulin91**
Some questions:
- Should we move check_b0_threshold from the script to the function?
- Should we leave parameter checks/warnings in the script? Both?

### CODE REVIEW

**Comment by ppoulin91** [comment]
For context: this PR is the first of many where we try to separate scripts from functions that we can import in the new scil_vital/dwi_ml repository.

**Comment by EmmaRenauld** [comment]
You miss this at the beginning

#!/usr/bin/env python
# -*- coding: utf-8 -*-

**Comment by boiteaclou** [comment]
This is my opinion, but I think that if the function should not be executed when the check don't pass, then the checks should be in the function and raise exceptions. Those exceptions should then be caught in the script for the logging messages. Else, if it is just on the format of inputs, then it should be only checked in the script.

**Comment by ppoulin91** [comment]
@EmmaRenauld I added the UTF-8 tag, but not the shebang since it's not a script

**Comment by AntoineTheb** [comment]
@boiteaclou Seems about right. imo the scripts should have as little "logic" in them as possible and only validate file inputs that would be otherwise already validated by other scripts also using the "logic" behind

**Comment by arnaudbore** [comment]
I'm not convinced it should end into a folder named `dwi` but everything else LGTM.

**Comment by ppoulin91** [comment]
Any other ideas for the folder name? It doesn't really fit with existing ones...

**Comment by AntoineTheb** [comment]
Could it fit in the `reconst` folder ?

**Comment by GuillaumeTh** [comment]
I agree with @AntoineTheb, it could fit in recont and maybe you can create a file 'frf.py'. Normally in the future we will have the multi-tissue response function.

**Comment by jchoude** [comment]
Yes, `reconst` would be a good module for that.

I also agree with @boiteaclou about the checks on files vs checks on arguments.

**Comment by ppoulin91** [comment]
Nothing else should change, anyone wants to review the last modifications before merging?

**Comment by arnaudbore** [comment]
If you change check_b0_threshold you have to do the modification everywhere.

**Comment by ppoulin91** [comment]
@arnaudbore I did, the only other usages were in compute_dti_metrics and compute_fodf

File: scilpy/reconst/frf.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+from dipy.segment.mask import applymask
+
+from scilpy.utils.bvec_bval_tools import (is_normalized_bvecs, normalize_bvecs,
+                                          check_b0_threshold)
+
+
+def compute_ssft_frf(data, bvals, bvecs, mask=None, mask_wm=None,
+                     fa_thresh=0.7, min_fa_thresh=0.5, min_nvox=300,
+                     roi_radius=10, roi_center=None, force_b0_threshold=False):
+    """Compute a single Fiber Response Function from a DWI.
```
Comment (jchoude) [inline scilpy/reconst/frf.py]: Maybe specify that this is for single shell data (or, at least, under b=1500). That the reason it's `ssst` response.

File: scilpy/reconst/frf.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import numpy as np
+from dipy.core.gradients import gradient_table
+from dipy.reconst.csdeconv import auto_response
+from dipy.segment.mask import applymask
+
+from scilpy.utils.bvec_bval_tools import (is_normalized_bvecs, normalize_bvecs,
+                                          check_b0_threshold)
+
+
+def compute_ssft_frf(data, bvals, bvecs, mask=None, mask_wm=None,
```
Comment (jchoude) [inline scilpy/reconst/frf.py]: In fact, should be `ssst`: single shell single tissue.

File: scilpy/reconst/frf.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    if mask_wm is not None:
+        data = applymask(data, mask_wm)
+    else:
+        logging.warning(
+            'No white matter mask specified! Only mask will be used '
+            '(if it has been supplied). \nBe *VERY* careful about the '
+            'estimation of the fiber response function to ensure no invalid '
+            'voxel was used.')
+
+    # Iteratively trying to fit at least 300 voxels. Lower the FA threshold
```
Comment (jchoude) [inline scilpy/reconst/frf.py]: 300 -> `min_nvox`

File: scilpy/reconst/frf.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    # when it doesn't work. Fail if the fa threshold is smaller than
+    # the min_threshold.
+    # We use an epsilon since the -= 0.05 might incurs numerical imprecision.
```
Comment (jchoude) [inline scilpy/reconst/frf.py]: `incurs` -> `incur`

File: scilpy/reconst/frf.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+import logging
+import numpy as np
```
Comment (jchoude) [inline scilpy/reconst/frf.py]: Theoretically, there should be a blank line between system and third party imports.

File: scilpy/reconst/frf.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    gtab = gradient_table(bvals, bvecs, b0_threshold=bvals.min())
+
+    if mask is not None:
+        data = applymask(data, mask)
+
+    if mask_wm is not None:
+        data = applymask(data, mask_wm)
+    else:
+        logging.warning(
+            'No white matter mask specified! Only mask will be used '
```
Comment (jchoude) [inline scilpy/reconst/frf.py]: There is a mix of `' '` and `" "` in the strings through the file. Should we uniformize?

File: scilpy/reconst/frf.py (hunk 2 above)
Comment (ppoulin91) [reply scilpy/reconst/frf.py]: Woops, fixed

File: scilpy/reconst/frf.py (hunk 1 above)
Comment (ppoulin91) [reply scilpy/reconst/frf.py]: Fixed

File: scilpy/reconst/frf.py (hunk 3 above)
Comment (ppoulin91) [reply scilpy/reconst/frf.py]: Done

File: scilpy/reconst/frf.py (hunk 4 above)
Comment (ppoulin91) [reply scilpy/reconst/frf.py]: Done

File: scilpy/reconst/frf.py (hunk 6 above)
Comment (ppoulin91) [reply scilpy/reconst/frf.py]: I'll convert everything to double quotes

**Comment by ppoulin91** [comment]
JC's comments were addressed

**Pull request merged**